src/app/components/filebrowseruss/filebrowseruss.component.ts

```diff
@@ -220,53 +224,101 @@ export class FileBrowserUSSComponent implements OnInit, OnDestroy {//IFileBrowse
+          this.pasteFile(this.fileToCopyOrCut,this.rightClickedFile.path)
+        }}
+      );
+      this.rightClickPropertiesPanel.push(
+        { text: "Paste", action:() => { 
+          this.pasteFile(this.fileToCopyOrCut,this.path)
         }}
       );
     }
-    this.fileToCopy = rightClickedFile.path
+    this.fileToCopyOrCut = rightClickedFile.path
+    this.isCutOperation = true;
   }
 
-  pasteFile(filePath: string, rightClickedFile: any) {
+  pasteFile(filePath: string, destinationPath: any) {
     let pathAndName = filePath;
     let name = this.getNameFromPathAndName(pathAndName);
-    this.isLoading = true;
-    let copySubscription = this.ussSrv.copyFile(pathAndName,rightClickedFile.path + "/" + name)
-    .subscribe(
-      resp => {
-        this.isLoading = false;
-        this.snackBar.open('Copied: ' + name,'Dismiss', { duration: 5000,   panelClass: 'center' });
-      },
-      error => {
-        this.ussSrv.deleteFileOrFolder(rightClickedFile.path + "/" + name).subscribe(() => {
-          if (error.status == '500') { //Internal Server Error
-            this.snackBar.open('Failed to copy: ' + pathAndName + "' This is probably due to a server agent problem.", 
-            'Dismiss', { duration: 5000,   panelClass: 'center' });
-          } else if (error.status == '404') { //Not Found
-            this.snackBar.open(pathAndName + ' does not exist.', 
-            'Dismiss', { duration: 5000,   panelClass: 'center' });
-          } else if (error.status == '400') { //Bad Request
-            this.snackBar.open("Failed to copy '" + pathAndName + "' This is probably due to a permission problem.", 
-            'Dismiss', { duration: 5000,   panelClass: 'center' });
-          } else { //Unknown
-            this.snackBar.open("Uknown error '" + error.status + "' occured for: " + pathAndName, 
+    if(this.getPathFromPathAndName(filePath) == destinationPath){
+      this.snackBar.open("Paste operation failed: '" + filePath + "' Cannot paste file to same destination.",'Dismiss', { duration: 5000, panelClass: 'center' });
+      return;
+    }
+    if(pathAndName.indexOf(' ') >= 0){
+      this.snackBar.open("Paste operation failed: '" + filePath + "' Paste operation not supported for filenames with spaces.",'Dismiss', { duration: 5000, panelClass: 'center' });
+      return;
+    }
+    let metaData = this.ussSrv.getFileMetadata(pathAndName);
+    metaData.subscribe(result => {
+      if(result.ccsid == -1){
+        this.snackBar.open("Paste operation failed: '" + filePath + "' Operation not supported.", 
             'Dismiss', { duration: 5000,   panelClass: 'center' });
+        return;
+      }else{
+        this.isLoading = true;
+        let copySubscription = this.ussSrv.copyFile(pathAndName,destinationPath + "/" + name)
+        .subscribe(
+          resp => {
+            this.isLoading = false;
+            this.updateUss(destinationPath);
+            if(this.isCutOperation){
```

Comment:
I'm a bit worried about the boolean being a instance variable in case there is some flow where it doesnt get reset properly. So, can this be a function argument instead?

Reply:
Added an argument to function and removed the global variable